pkg/runs/resultParameterChecker.go

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package runs
+
+import (
+	"context"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/galasa.dev/cli/pkg/api"
+	"github.com/galasa.dev/cli/pkg/errors"
+)
+
+func ValidateResultParameter(resultInputString string, apiServerUrl string) (string, error) {
+	var err error = nil
+	var validResultInputs []string
+	var invalidResultInputs []string
+	var resultQuery string = ""
+
+	restClient := api.InitialiseAPI(apiServerUrl)
+	var context context.Context = nil
+	rasResultNamesData, httpResponse, err := restClient.ResultArchiveStoreAPIApi.GetRasResultNames(context).Execute()
+
+	if err == nil {
+		if httpResponse.StatusCode != http.StatusOK {
+			httpError := "\nhttp response status code: " + strconv.Itoa(httpResponse.StatusCode)
+			errString := err.Error() + httpError
+			err = errors.NewGalasaError(errors.GALASA_ERROR_QUERY_RESULTNAMES_FAILED, errString)
+		} else {
+			rasResultNames := rasResultNamesData.GetResultnames()
+			resultInputs := strings.Split(resultInputString, ",")
+
+			// compare the input result to the list of possibles - case insensitive
+			for _, resultInput := range resultInputs {
+				matched := false
+				for _, rasResultName := range rasResultNames {
```

Comment:
If you put the valid result names into a map[string], then you could just see if that string was in the map without this linear search. Simpler code ?